src/modules/site-settings/edit-site-details.tsx

```diff
@@ -214,8 +272,11 @@ export default function EditSiteDetails( { currentWpVersion, onSave }: EditSiteD
 										! selectedSite ||
 										( selectedSite?.name === siteName &&
 											selectedSite?.phpVersion === selectedPhpVersion &&
-											currentWpVersion === selectedWpVersion ) ||
-										! siteName.trim()
+											currentWpVersion === selectedWpVersion &&
```

Comment:
The disabled state logic is a bit long and could be made more readable by extracting it into meaningful variables:

```typescript
const isFormUnchanged = selectedSite && 
    selectedSite.name === siteName &&
    selectedSite.phpVersion === selectedPhpVersion &&
    currentWpVersion === selectedWpVersion &&
    Boolean(selectedSite.customDomain) === useCustomDomain &&
    (!useCustomDomain || selectedSite.customDomain === customDomain);

const hasValidationErrors = !selectedSite || 
    !siteName.trim() || 
    (useCustomDomain && customDomainError);

disabled={isEditingSite || isFormUnchanged || hasValidationErrors}
```

This would make the code more self-documenting and easier to maintain. What do you think?

Reply:
Sounds good to me :)